tests/conftest.py

```diff
@@ -478,5 +504,38 @@ def wrap_training_methods() -> None:
         training_module.train = raise_on_unexpected_train(training_module.train)
 
 
-def pytest_configure():
+def pytest_configure() -> None:
     wrap_training_methods()
+
+
+def _get_marker_for_ci_matrix(item: Function) -> Text:
+    """Returns pytest marker which is used to parallelize the tests in GitHub actions.
+
+    Args:
+        item: The test case.
+
+    Returns:
+        A marker for this test based on which directory / python module the test is in.
+    """
+    test_path = Path(item.fspath).absolute()
+
+    for marker, paths_for_marker in PATH_PYTEST_MARKER_MAPPINGS.items():
+        if any(
+            path == test_path or path in test_path.parents for path in paths_for_marker
+        ):
+            return marker
```

Comment:
Maybe it's worth checking and raising if it matches two items.

Reply:
done